fdbserver/ptxn/test/TestTLogServer.actor.cpp

```diff
@@ -848,6 +834,116 @@ TEST_CASE("/fdbserver/ptxn/test/read_persisted_disk_on_tlog") {
+		for (ptxn::TLogGroup& tlogGroup : pContext->groupsPerTLog[i]) {
+			std::string filename =
+			    filenameFromId(req.storeType, folder, prefix.toString() + "test", tlogGroup.logGroupId);
+			IKeyValueStore* data = openKVStore(req.storeType, filename, tlogGroup.logGroupId, 500e6);
+			state IDiskQueue* queue = new InMemoryDiskQueue(tlogGroup.logGroupId);
+			ds[tlogGroup.logGroupId] = data;
+			persistentDataAndQueues[tlogGroup.logGroupId] = std::make_pair(data, queue);
+		}
+
+		actors.push_back(ptxn::tLog(persistentDataAndQueues,
+		                            makeReference<AsyncVar<ServerDBInfo>>(),
+		                            LocalityData(),
+		                            initializeTLog,
+		                            tlogId,
+		                            workerId,
+		                            false,
+		                            Promise<Void>(),
+		                            Promise<Void>(),
+		                            folder,
+		                            makeReference<AsyncVar<bool>>(false),
+		                            makeReference<AsyncVar<UID>>(tlogId)));
+		initializeTLog.send(tLogInitializations.back());
+		std::cout << "Recruit tlog " << i << " : " << tlogId.shortString() << ", workerID: " << workerId.shortString()
+		          << "\n";
+	}
+
+	// replace fake TLogInterface with recruited interface
+	std::vector<Future<ptxn::TLogInterface_PassivelyPull>> interfaceFutures(pContext->numTLogs);
+	for (i = 0; i < pContext->numTLogs; i++) {
+		interfaceFutures[i] = tLogInitializations[i].reply.getFuture();
+	}
+	std::vector<ptxn::TLogInterface_PassivelyPull> interfaces = wait(getAll(interfaceFutures));
+	for (i = 0; i < pContext->numTLogs; i++) {
+		*(pContext->tLogInterfaces[i]) = interfaces[i];
+	}
+
+	for (auto& [tLogGroupID, tLogGroupLeader] : pContext->tLogGroupLeaders) {
+		tLogGroupLeader = pContext->tLogInterfaces[groupToLeaderId[tLogGroupID]];
+	}
+
+	state IKeyValueStore* d = ds[pContext->storageTeamIDTLogGroupIDMapper[storageTeamID]];
+
+	state std::pair<std::vector<Standalone<StringRef>>, std::vector<Version>> res =
+	    wait(commitInjectReturnVersions(pContext, storageTeamID, pContext->numCommits));
+	state std::vector<Standalone<StringRef>> expectedMessages = res.first;
+	wait(verifyPeek(pContext, storageTeamID, pContext->numCommits));
+
+	// wait 1s so that actors who update persistent data can do their job.
+	wait(delay(1.5));
+
+	// only wrote to a single storageTeamId, thus only 1 tlogGroup, while each tlogGroup has their own disk queue.
+	state bool exist = false;
+	// commit to IKeyValueStore might happen in any version of our commits(might happen more than time)
+	for (i = 0; i < res.second.size(); i++) {
+		state Key k = ptxn::persistStorageTeamMessageRefsKey(
+		    pContext->getTLogLeaderByStorageTeamID(storageTeamID)->id(), storageTeamID, res.second[i]);
+		state Optional<Value> v = wait(d->readValue(k));
+		exist = exist || v.present();
```

Comment:
Since spill threshold is 0. Does it make sense to `ASSERT(v.present())` here? It's slightly stronger than checking if any exists.

Reply:
it would not work, spilling does not always happen for each version, see my comment `// commit to IKeyValueStore might happen in any version of our commits(might happen more than time)`

Reply:
Got it. Make sense.